api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/RepeatVectorLayerTest.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlinx.dl.api.core.layer
+
+import org.jetbrains.kotlinx.dl.api.core.KGraph
+import org.jetbrains.kotlinx.dl.api.core.layer.reshaping.RepeatVector
+import org.jetbrains.kotlinx.dl.api.core.shape.toIntArray
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.tensorflow.Graph
+import org.tensorflow.Output
+import org.tensorflow.Shape
+import org.tensorflow.op.Ops
+
+internal class RepeatVectorLayerTest {
+
+    @Test
+    fun `test output shape`() {
+        val layer = RepeatVector(n = 2)
+        val x = Array(10) { FloatArray(10) { 1F } }
+        val y = layer(x)
+        Assertions.assertArrayEquals(intArrayOf(10, layer.n, 10), y.shape().toIntArray())
+    }
+
+    @Test
+    fun `test repetition output`() {
+        val layer = RepeatVector(n = 2)
+        val x = Array(3) { FloatArray(3) { it.toFloat() } }
+        val y = layer(x)
+        val actual = y.tensor().copyTo(Array(3) { Array(layer.n) { FloatArray(3) } })
+        val expected = arrayOf(
+            arrayOf(floatArrayOf(0F, 1F, 2F), floatArrayOf(0F, 1F, 2F)),
+            arrayOf(floatArrayOf(0F, 1F, 2F), floatArrayOf(0F, 1F, 2F)),
+            arrayOf(floatArrayOf(0F, 1F, 2F), floatArrayOf(0F, 1F, 2F))
+        )
+        Assertions.assertArrayEquals(expected, actual)
+    }
+
+    // TODO: generalise this for Layer, see https://github.com/JetBrains/KotlinDL/issues/145
+    private operator fun RepeatVector.invoke(input: Array<FloatArray>): Output<Float> = Ops.create().let { tf ->
```

Comment:
maybe you could make a util internal extension function for the Layer class?

Reply:
maybe I could, u mean as described in #145 ? 
What should I do for now though, remove the function?

Reply:
So at this moment, you could leave it as is, I suppose.
The improvement related to #145 should be implemented in the separate PR and discussed with other contributors faced with this problem.